src/test/regress/expected/multi_fix_partition_shard_index_names.out

```diff
@@ -564,74 +469,171 @@ SELECT create_distributed_table('dist_partitioned_table', 'dist_col');
 ---------------------------------------------------------------------
- dist_partitioned_table_910030              | short_910030
- p_910032                                   | p_another_col_partition_col_idx_910032
- partition_table_with_very_long_name_910031 | partition_table_with_very_long_na_another_col_p_dd884a3b_910031
-(3 rows)
 
-\c - - - :master_port
-SET search_path TO fix_idx_names, public;
--- should now work
-ALTER INDEX partition_table_with_very_long_na_another_col_partition_col_idx RENAME TO partition_table_with_very_long_name_idx;
--- now we can drop index on detached partition
-ALTER TABLE dist_partitioned_table DETACH PARTITION p;
-DROP INDEX p_another_col_partition_col_idx;
-\c - - - :worker_1_port
--- check that indexes have been renamed
--- and that index on p has been dropped (it won't appear)
+(1 row)
+
+CREATE TABLE partition_local_table PARTITION OF date_partitioned_citus_local_table FOR VALUES FROM ('2018-01-01') TO ('2019-01-01');
+-- SELECT fix_partition_shard_index_names('date_partitioned_citus_local_table') will be executed
+-- automatically at the end of the CREATE INDEX command
+CREATE INDEX ON date_partitioned_citus_local_table USING btree(measureid);
+-- check that index names are correct
 SELECT tablename, indexname FROM pg_indexes WHERE schemaname = 'fix_idx_names' ORDER BY 1, 2;
-                 tablename                  |                   indexname
+                 tablename                 |                        indexname
 ---------------------------------------------------------------------
- dist_partitioned_table_910030              | short_910030
- partition_table_with_very_long_name_910031 | partition_table_with_very_long_name_idx_910031
-(2 rows)
+ date_partitioned_citus_local_table        | date_partitioned_citus_local_table_measureid_idx
+ date_partitioned_citus_local_table_361377 | date_partitioned_citus_local_table_measureid_idx_361377
+ partition_local_table                     | partition_local_table_measureid_idx
+ partition_local_table_361378              | partition_local_table_measureid_idx_361378
+(4 rows)
 
-\c - - - :master_port
-SET search_path TO fix_idx_names, public;
+-- creating a single object should only need to trigger fixing the single object
+-- for example, if a partitioned table has already many indexes and we create a new
+-- index, only the new index should be fixed
+-- create only one shard & one partition so that the output easier to check
+SET citus.next_shard_id TO 915000;
+SET citus.shard_count TO 1;
+SET citus.shard_replication_factor TO 1;
+CREATE TABLE parent_table (dist_col int, another_col int, partition_col timestamp, name text) PARTITION BY RANGE (partition_col);
+SELECT create_distributed_table('parent_table', 'dist_col');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE p1 PARTITION OF parent_table FOR VALUES FROM ('2018-01-01') TO ('2019-01-01');
```

Comment:
I wish we had https://github.com/citusdata/citus/pull/4634 so that we can only filter `worker_fix_partition_shard_index_names`